x-pack/plugins/apm/public/services/rest/ml.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ESFilter } from 'elasticsearch';
+import chrome from 'ui/chrome';
+import { SERVICE_NAME, TRANSACTION_TYPE } from '../../../common/constants';
+import { callApi } from './callApi';
+
+export function getMlPrefix(serviceName: string, transactionType?: string) {
+  return `${serviceName}-${
+    transactionType ? transactionType + '-' : ''
+  }`.toLowerCase();
+}
+
+interface MlResponseItem {
+  id: string;
+  success: boolean;
+  error?: {
+    msg: string;
+    body: string;
+    path: string;
+    response: string;
+    statusCode: number;
+  };
+}
+
+interface StartMLJobApiResponse {
+  datafeeds: MlResponseItem[];
+  jobs: MlResponseItem[];
+}
+
+export async function startMLJob({
+  serviceName,
+  transactionType
+}: {
+  serviceName: string;
+  transactionType?: string;
+}) {
+  const indexPatternName = chrome.getInjected('apmIndexPatternTitle');
+  const groups = ['apm', serviceName.toLowerCase()];
+  const filter: ESFilter[] = [{ term: { [SERVICE_NAME]: serviceName } }];
+  if (transactionType) {
+    groups.push(transactionType.toLowerCase());
+    filter.push({ term: { [TRANSACTION_TYPE]: transactionType } });
+  }
+  return callApi<StartMLJobApiResponse>({
+    method: 'POST',
+    pathname: `/api/ml/modules/setup/apm_transaction`,
+    body: JSON.stringify({
+      prefix: getMlPrefix(serviceName, transactionType),
+      groups,
+      indexPatternName,
+      startDatafeed: true,
+      query: {
+        bool: {
+          filter
+        }
+      }
+    })
+  });
+}
+
+// https://www.elastic.co/guide/en/elasticsearch/reference/6.5/ml-get-job.html
+export interface GetMLJobApiResponse {
```

Comment:
Nit: For the other interfaces we haven't prefixed with a verb (start, get). To make it consistent we could name them like:

```
StartMLJobApiResponse -> StartedMLJobApiResponse
GetMLJobApiResponse -> MLJobApiResponse
```

I think that makes sense if you think about what is returned vs what action caused it.